apps/emqx_auth_http/src/emqx_auth_http.erl

```diff
@@ -83,7 +83,7 @@ authenticate(#http_request{url = Url,
              ClientInfo, HttpHeaders, HttpOpts, RetryOpts) ->
    request(Method, ContentType, Url, feedvar(Params, ClientInfo), HttpHeaders, HttpOpts, Options, RetryOpts).
 
--spec(is_superuser(maybe(#http_request{}), emqx_types:client(), list(), list(), list()) -> boolean()).
+-spec(is_superuser(undefined | #http_request{method::'get' | 'post',params::[any()]}, #{'clientid':=_, 'peerhost':=_, 'protocol':=_, _=>_}, Headers::any(), HttpOpts::any(), #{'backoff':=_, 'interval':=_, 'times':=_, _=>_}) -> boolean()).
```

Comment:
please add a type spec for the record in `emqx_auth_http.hrl`

Reply:
sorry, adding type spec in hrl is not recommended, please add the type spec in this module and use it here.